server/src/main/java/org/elasticsearch/search/aggregations/CardinalityUpperBound.java

```diff
@@ -47,35 +54,32 @@ public CardinalityUpperBound multiply(int bucketCount) {
      * aggregations like {@link FilterAggregator} or a {@link RangeAggregator}
      * configured to collect only a single range.
      */
-    ONE {
-        @Override
-        public CardinalityUpperBound multiply(int bucketCount) {
-            switch (bucketCount) {
-                case 0:
-                    return NONE;
-                case 1:
-                    return ONE;
-                default:
-                    return MANY;
-            }
-        }
-    },
+    public static final CardinalityUpperBound ONE = new KnownCardinalityUpperBound(1);
 
     /**
      * {@link Aggregator}s with this cardinality may be collected many times.
      * Most sub-aggregators of {@link BucketsAggregator}s will have
      * this cardinality.
      */
-    MANY {
+    public static final CardinalityUpperBound MANY = new CardinalityUpperBound() {
         @Override
         public CardinalityUpperBound multiply(int bucketCount) {
             if (bucketCount == 0) {
                 return NONE;
             }
             return MANY;
         }
+
+        @Override
+        public <R> R map(IntFunction<R> mapper) {
+            return mapper.apply(Integer.MAX_VALUE);
```

Comment:
When I first wrote this I had map take two parameters - one for if the value was unknown and one for if it was known. And that turned out to be pretty inconvenient. In the cases we have "very big" might as well be unknown. So pegging unknown as "the biggest int" seems to work well here.